src/Swarm/Language/Syntax.hs

```diff
@@ -404,6 +412,15 @@ data MUnAssoc
     S
   deriving (Eq, Ord, Show)
 
+-- | Whether a command is internal or external.  External
+--   commands have some kind of effect on the external world; at most
+--   one external command can be executed per tick.  Internal commands
+--   are things like sensing commands, or commands that solely modify
+--   a robot's internal state; multiple internal commands may be
+--   executed per tick.
+data Facing = Internal | External
```

Comment:
I would suggest _silent_ vs _observable_ visibility (found it on the [bisimulation Wiki page](https://en.wikipedia.org/wiki/Bisimulation#Variants_of_bisimulation)) so that it's not as easy to confuse with other things we keep internal like the `CGod` commands.

Reply:
Hmm, I agree internal/external is confusing.  Silent/observable don't really sound like opposites to me, so I consulted a thesaurus and went with "tangible/intangible" --- how does that sound to you?